Cocoapods/Networking/Requests/Abstract/CCPRequest.m

```diff
@@ -0,0 +1,30 @@
+//
+//  CCPRequest.m
+//  Cocoapods
+//
+//  Created by Gabriel Massana on 7/2/16.
+//  Copyright © 2016 Gabriel Massana. All rights reserved.
+//
+
+#import "CCPRequest.h"
+
+@implementation CCPRequest
+
++ (instancetype)requestForAPI
+{
+    CCPRequest *request = [self request];
+    
+    return request;
+}
+
+- (void)updateRequestWithEndpoint:(NSString *)endpoint
```

Comment:
This method is doing almost the same as our - (void)setEndpoint:(NSString *)endpoint;
But I changed the name to not be set.
The function of the method is to merge the constant part of the API URL with the enpoint of this request.

Please, tell me if you can imagine a better name, I'll be happy to change it.